### PR TITLE
Release/search filter

### PR DESCRIPTION
**Pull request by zoyar-moveo**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,7 +1,20 @@
 import "./App.css";
+import SearchFllter from "./components/SearchFilter/SearchFilter";
+import SearchIcon from "./components/SearchFilter/assets/search.svg";
+import ForwardIcon from "./components/SearchFilter/assets/Forward.svg";
 
 function App() {
-  return <></>;
+  return (
+    <>
+      <SearchFllter
```
Comment (noam-movhls) [inline src/App.tsx]: please remove if it's redundant

File: src/components/DataCard/charts/sourcesChart/SourcesChart.tsx [hunk 2]
```diff
@@ -4,7 +4,7 @@ import * as chartjs from "chart.js";
 
 type dataSetType = {
   data: number[];
-  backgroundColor: string[];
+  backgroundColor: any;
```
Comment (noam-movhls) [inline src/components/DataCard/charts/sourcesChart/SourcesChart.tsx]: please read here about CSS properties

https://stackoverflow.com/questions/50960084/how-to-extends-cssproperties-in-react-project

maybe you will find it useful

File: src/components/FilterList/FilterList.tsx [hunk 3]
```diff
@@ -12,6 +12,7 @@ const FilterList: React.FC<FilterListProps> = (props) => {
       {props.filterData.map((dataType: string, index: number) => {
         return (
           <Filter
+            filterSort="secondary"
```
Comment (noam-movhls) [inline src/components/FilterList/FilterList.tsx:15]: please consider to make it type/enum

File: src/components/SearchFilter/SearchFilter.tsx [hunk 4]
```diff
@@ -1,17 +1,112 @@
...
 const SearchFllter: React.FC<SearchFilterProps> = (props) => {
+  const [isDropDownOpen, setIsDropDownOpen] = useState(false);
+  const [searchInput, setSearchInput] = useState("");
+  const [currFilter, setCurrFilter] = useState("All");
+  const inputRef: any = useRef();
+
+  const updateFilter = (currFilter: string) => {
+    setCurrFilter(currFilter);
+    setIsDropDownOpen(false);
+    // parentUpdate(currFilter);
```
Comment (noam-movhls) [inline src/components/SearchFilter/SearchFilter.tsx]: remove if redundant

File: src/components/SearchFilter/SearchFilter.tsx [hunk 5]
```diff
@@ -1,17 +1,112 @@
...
+    ev.preventDefault();
+    let val = ev.target.value;
+    setSearchInput(val);
+  };
+
+  const toggleResentSearches = () => {
+    setIsDropDownOpen((state) => !state);
+  };
+
+  // const innerFilter = {
```
Comment (noam-movhls) [inline src/components/SearchFilter/SearchFilter.tsx]: remove if redundant

File: src/components/SearchFilter/SearchFilter.tsx [hunk 6]
```diff
@@ -1,17 +1,112 @@
...
+  SearchsList: string[];
+  searchIcon: string;
+  forwardIcon: string;
 }
 
 const SearchFllter: React.FC<SearchFilterProps> = (props) => {
+  const [isDropDownOpen, setIsDropDownOpen] = useState(false);
+  const [searchInput, setSearchInput] = useState("");
+  const [currFilter, setCurrFilter] = useState("All");
+  const inputRef: any = useRef();
```
Comment (noam-movhls) [inline src/components/SearchFilter/SearchFilter.tsx]: type React.node is not good enough;?

File: src/components/SearchFilter/SearchFilter.tsx [hunk 7]
```diff
@@ -1,17 +1,112 @@
...
+  // };
+
+  const parentUpdate = (filter: string) => console.log(filter);
+
+  useEffect(() => {}, [isDropDownOpen]);
   return (
     <>
       <GlobalStyles />
-      <FilterContainer></FilterContainer>
+      {/* <SearchFilterContainer> */}
```
Comment (noam-movhls) [inline src/components/SearchFilter/SearchFilter.tsx]: same here

File: src/components/SearchFilter/SearchFilter.tsx [hunk 8]
```diff
@@ -1,17 +1,112 @@
...
+          <FlexSpaceBetween>
+            <SearchContainer>
+              <Img alt="" src={props.searchIcon} />
+              <input
+                type="text"
+                value={searchInput}
+                placeholder="search"
+                onChange={inputUpdate}
+                onFocus={toggleResentSearches}
+                // ref={inputRef}
```
Comment (noam-movhls) [inline src/components/SearchFilter/SearchFilter.tsx]: is it necessary?

File: src/components/SearchFilter/SearchFilter.tsx (hunk 6 above)
Comment (zoyar-moveo) [reply src/components/SearchFilter/SearchFilter.tsx]: useRef<HTMLInputElement | null> did the trick

File: src/components/FilterList/FilterList.tsx (hunk 3 above)
Comment (zoyar-moveo) [reply src/components/FilterList/FilterList.tsx:15]: Added task to Chores

**Pull request merged**